projects/remove-all-whitespace/index.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Remove All Whitespace in Every Language
+layout: default
+date: 2020-10-21
+last-modified: 2020-10-21
+featured-image:
+tags: [removingExtraWhiteSpaces, c]
+authors:
+    - [BHMBS Ibrahim](https://github.com/barhouum7)
+---
+
+Removing extra white spaces from a given string
+
+Given a string, remove all spaces from the string and return it with it's _length_.
+
+```c
+Input:  " Removing   Extra White   Spaces  "
+Output: "removingExtraWhiteSpaces"
+```
+
+## Requirements
+
+A string is nothing but an array of characters. The value of a string is determined by the terminating character. Its value is considered to be 0.
+
+As given in the example above, you need to enter a string first up.
+
+The string entered here is as follows: "&nbsp;&nbsp;Removing&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extra&nbsp;White&nbsp;&nbsp;&nbsp;Spaces&nbsp;&nbsp;&nbsp;&nbsp;"
+
+As you can see, there are quite a few blank spaces in the string entered above.
+
+Hence, the string becomes like this after removing all the blank spaces:
+
+"removingExtraWhiteSpaces"
+
+## Testing
+
+The following table contains various test cases that you can use to verify the 
+correctness of your solution:
+
+| Description | Input | Output |
+|--------------|-------|--------|
+| No Input | | "Wrong input! , please try again." |
+| Empty Input | "" | "Wrong input! , please try again." |
+| Any String Input| "&nbsp;&nbsp;Removing&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Extra&nbsp;White&nbsp;&nbsp;&nbsp;Spaces&nbsp;&nbsp;&nbsp;&nbsp;" | "removingExtraWhiteSpaces" |
```

Comment:
```suggestion
| Sample Input: Leading Spaces | "      RemoveExtraWhitespace" | "RemoveExtraWhitespace" |
```

This was a suggestion for a simpler test case. Could we make one for trailing and one for inner spaces?

Reply:
What do you mean about making one for trailing and one for inner spaces?

Reply:
Also, can you please add the hacktoberfest label to this PR so that I will become eligible for the prize?

Reply:
This first example has spaces in the front. It would be cool to have one that has spaces after the string (trailing) and spaces between the words (inner). I think those would be three good tests for a program.

Reply:
> This first example has spaces in the front. It would be cool to have one that has spaces after the string (trailing) and spaces between the words (inner). I think those would be three good tests for a program.

Do you want to add another simple input for trailing and inner tests?